File: apps/SleepBluetooth/src/org/lineageos/atv/sleepbt/BootReceiver.java
/*
 * SPDX-FileCopyrightText: The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.lineageos.atv.sleepbt;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/** Starts {@link SleepBluetoothService} once the device finishes booting. */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            context.startService(new Intent(context, SleepBluetoothService.class));
        }
    }
}

File: apps/ATVPackageDisabler/src/org/lineageos/atvpackagedisabler/BootReceiver.kt
/*
 * SPDX-FileCopyrightText: The LineageOS Project
 * SPDX-License-Identifier: Apache-2.0
 */
package org.lineageos.atvpackagedisabler

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Log.d(TAG, "Received boot completed intent")
        PackageDisabler.disablePackages(context)
    }

    companion object {
        private const val TAG = "BootReceiver"
    }
}

